src/coreclr/jit/sideeffects.cpp

```diff
@@ -140,14 +140,34 @@ AliasSet::NodeInfo::NodeInfo(Compiler* compiler, GenTree* node)
 {
     if (node->IsCall())
     {
+        // For calls having return buffer, update the local number that is written after his call.
+        GenTree* retBufArgNode = node->AsCall()->GetLclRetBufArgNode();
+        if (retBufArgNode != nullptr)
+        {
+            if (retBufArgNode->OperIsPutArg())
+            {
+                retBufArgNode = retBufArgNode->AsOp()->gtGetOp1();
+            }
```

Comment:
I think it would be better to check for the putarg inside `GetLclRetBufArgNode`. It should also check for `STORE_LCL_VAR` (to mirror the `ASG` setup node check).

Reply:
> I think it would be better to check for the putarg inside GetLclRetBufArgNode.
Done.

>  It should also check for STORE_LCL_VAR (to mirror the ASG setup node check).

Hhm, it sets `isWrite = true` and then sets `ALIAS_WRITES_ADDRESSABLE_LOCATION` , `ALIAS_WRITES_LCL_VAR` which we are already doing?

Reply:
> Hhm, it sets `isWrite = true` and then sets `ALIAS_WRITES_ADDRESSABLE_LOCATION` , `ALIAS_WRITES_LCL_VAR` which we are already doing?

The code in `sideffects.cpp` is correct -- I mean that inside `GetLclRetBufArgNode` we check for `ASG` and if there is one take it's RHS as the value for the return. Seeing as those `ASG`s get transformed into `STORE_LCL_VAR`s (always local because the setup arg always assigns to a temp) in the backend, we should also check for that.